client/src/utils/Store.js

```diff
@@ -195,6 +216,24 @@ export default class API {
    */
   getModel = async (name) => clonedeep(await this?.state.models[name]);
 
+  /**
+   * Gets (clones) all models.
+   *
+   * @returns {{
+   *  [modelName: String]: Object
+   * } | undefined} A copy of all the models with their names or undefined
+   */
+  getModels = async () => {
+    const modelNames = this.getModelNames();
+    const modelPromises = modelNames.map((modelName) => this?.state.models[modelName]);
+    const allModelObjs = clonedeep(await Promise.all(modelPromises));
+    const allModels = {};
+    modelNames.forEach((modelName, index) => {
+      allModels[modelName] = allModelObjs[index];
+    });
+    return allModels;
+  };
```

Comment:
Would be good if this was called getAllModels just to be more explicit.

Can you explain what the foreach is doing here and why you can't just return allModelObjs? Maybe included as comments inline? Assuming its related to models being stored as promises in the current setup of the store but would be good to have documentation for it just for transparency.

Reply:
@Tengelma having promises in the store is exactly why. Will add a comment about it now.